client/src/components/History/CurrentCollection/CollectionPanel.vue

```diff
@@ -59,9 +59,16 @@ function onScroll(newOffset: number) {
     offset.value = newOffset;
 }
 
-async function onViewSubCollection(itemObject: DCObject) {
-    const collection = await collectionElementsStore.getCollection(itemObject.id);
-    emit("view-collection", collection);
+async function onViewSubCollection(itemObject: DCObject, name: string) {
+    // We need to convert the DCO to a SubCollection by providing
```

Comment:
Would it make more sense to call this onViewDatasetCollectionElement ? That lines it up with the internal state and the objects coming out of the API. That would also mean you would pass around individual DatasetCollectionElements, which would avoid having to pass around `name`.

Reply:
Done! now we pass directly the DatasetCollectionElement instead of the internal DCObject.